rdrf/rdrf/clinician_view.py

```diff
@@ -0,0 +1,260 @@
+
+        return cleaned_data
+
+    def _get_users_queryset(self, registry_model):
+        try:
+            clinicians_group = Group.objects.get(name="Clinical Staff")
+            return CustomUser.objects.filter(
+                registry__in=[
+                    self.registry_model],
+                groups__in=[clinicians_group])
+        except Group.DoesNotExist:
+            return CustomUser.objects.none()
+
+    def _option_from_instance(self, clinician_user):
+        if clinician_user.working_groups.count() == 1:
+            working_group_names = clinician_user.working_groups.first().name
+        else:
+            working_group_names = ",".join(
+                [wg.name for wg in clinician_user.working_groups.all()])
+
+        option_string = "%s %s (%s)" % (clinician_user.first_name,
+                                        clinician_user.last_name,
+                                        working_group_names)
+
+        return [clinician_user.pk, option_string]
+
+
+class ClinicianFormView(View):
+    def _get_template(self):
+        return 'rdrf_cdes/clinician.html'
+
+    @method_decorator(login_required)
+    def get(self, request, registry_code, patient_id):
+
+        self._get_objects(request, registry_code, patient_id)
+
+        context = self._build_context()
+
+        return self._render_context(request, context)
+
+    def _build_context(self):
+        context = {
+            "location": "Clinician",
+            'patient_link': PatientLocator(
+                self.registry_model,
+                self.patient_model).link,
+            "previous_form_link": self.wizard.previous_link,
+            "next_form_link": self.wizard.next_link,
+            "form_name": _("Supervising Clinician"),
+            "registry_code": self.registry_model.code,
+            "patient_model": self.patient_model,
+            "parent": self.parent,
+            "form": self.clinician_form,
+            "context_launcher": self.context_launcher.html,
+        }
+
+        return context
+
+    def _get_navigation_wizard(self):
+        return NavigationWizard(self.request.user,
+                                self.registry_model,
+                                self.patient_model,
+                                NavigationFormType.CLINICIAN,
+                                None,
+                                None)
+
+    def _render_context(self, request, context):
+        context.update(csrf(request))
+        return render(request, self._get_template(), context)
+
+    def _get_objects(self, request, registry_code, patient_id):
+        self.request = request
+        self.user = request.user
+        self.message = None
+        if self.user.is_parent:
+            self.parent = ParentGuardian.objects.get(user=self.user)
+        else:
+            self.parent = None
+        self.patient_model = get_object_or_404(Patient, pk=patient_id)
+        if not security_check_user_patient(self.user, self.patient_model):
```

Comment:
I think if this function checks and enforces permissions it should be a bit clearer; or could this check be broken out and put somewhere more obvious? just so that someone doesn't lose it refactoring later on